build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/AndroidCompose.kt

```diff
@@ -53,48 +51,19 @@ internal fun Project.configureAndroidCompose(
         }
     }
 
-    tasks.withType<KotlinCompile>().configureEach {
-        kotlinOptions {
-            freeCompilerArgs += buildComposeMetricsParameters() 
-            freeCompilerArgs += stabilityConfiguration()
-            freeCompilerArgs += strongSkippingConfiguration()
-        }
-    }
-}
+    extensions.configure<ComposeCompilerGradlePluginExtension> {
+        fun Provider<String>.onlyIfTrue() = flatMap { provider { it.takeIf(String::toBoolean) } }
 
-private fun Project.buildComposeMetricsParameters(): List<String> {
-    val metricParameters = mutableListOf<String>()
-    val enableMetricsProvider = project.providers.gradleProperty("enableComposeCompilerMetrics")
-    val relativePath = projectDir.relativeTo(rootDir)
-    val buildDir = layout.buildDirectory.get().asFile
-    val enableMetrics = (enableMetricsProvider.orNull == "true")
-    if (enableMetrics) {
-        val metricsFolder = buildDir.resolve("compose-metrics").resolve(relativePath)
-        metricParameters.add("-P")
-        metricParameters.add(
-            "plugin:androidx.compose.compiler.plugins.kotlin:metricsDestination=" + metricsFolder.absolutePath,
-        )
-    }
+        project.providers.gradleProperty("enableComposeCompilerMetrics")
+            .onlyIfTrue().flatMap { rootProject.layout.buildDirectory.dir("compose-metrics") }
```

Comment:
If I am reading this right, this will put it in the rootProject build directory, where as before it would be in each modules build directory. I could be wrong though, so ignore if not true

Reply:
Good catch, but this is not exactly what was happening.
In fact they were all in the root directory, but they also kept their relative path: `/build/app/*`, `/build/core/analytics/*`, etc.

Let me re-add this behavior from: https://github.com/android/nowinandroid/pull/754

Reply:
Fixed in https://github.com/android/nowinandroid/pull/1036/commits/7bad454fdaf99167cdbbd3d4940488c4d3770b3c

Can you take a look and see if this works for you?
`./gradlew clean :app:assembleDemoDebug -PenableComposeCompilerMetrics=true -PenableComposeCompilerReports=true`

![image](https://github.com/android/nowinandroid/assets/1921278/3a78e6ca-8b16-4513-9464-ec91c647c961)

Reply:
That being said, we could also update the docs that are not very precise (in another PR?):

https://github.com/android/nowinandroid/blob/211654ff16aea732cab8fd6148998c2a7293612f/README.md?plain=1#L195-L196

```markdown
The reports files will be added to `build/[module]/compose-reports`. The metrics files will also be added to `build/[module]/compose-metrics`. 
```